atomic_reactor/plugins/pre_yum_proxy.py

```diff
@@ -0,0 +1,162 @@
+"""
+Copyright (c) 2019 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+from __future__ import absolute_import
+
+import os
+
+from six.moves.urllib.parse import urlsplit, urlunsplit
+from six.moves.configparser import ConfigParser
+
+from atomic_reactor.plugin import PreBuildPlugin
+from atomic_reactor.plugins.pre_reactor_config import get_yum_proxies
+from atomic_reactor.util import df_parser
+
+
+class YumProxyPlugin(PreBuildPlugin):
+    key = 'yum_proxy'
+    is_allowed_to_fail = False
+
+    def __init__(self, tasker, workflow):
+        """
+        Plugin initializer
+
+        :param tasker: DockerTasker instance
+        :param workflow: DockerBuildWorkflow instance
+        """
+        super(YumProxyPlugin, self).__init__(tasker, workflow)
+
+        # TODO: Is this dir accessbile in the build context?
+        self.workdir = os.path.join(self.workflow.source.workdir, 'atomic-reactor-yum-repos')
+        self.all_yum_repo_files = {}
+        self.updated_yum_repo_files = {}
+        self.yum_proxies = None
+        # TODO: Clean up containers and its volumes.
+        self._container_ids = []
+
+    def run(self):
+        """
+        run the plugin
+        """
+        try:
+            yum_proxies = get_yum_proxies(self.workflow)
+        except KeyError:
+            return
+        # Let's make it easier to search which hosts should be proxied.
+        self.yum_proxies = {info['proxied']: info['proxy'] for info in yum_proxies}
+
+        # Iterate through each parent image to find any hard coded yum repo file.
+        for parent_image_id in self.workflow.builder.parent_images.keys():
+            # TODO: will this ^ ignore SCRATCH_FROM ? Needs testing, but it looks like
+            # SCRATCH_FROM are exluded from parent_images .
+            self.all_yum_repo_files[parent_image_id] = self._get_yum_repos(parent_image_id)
+
+        self._generate_new_yum_repo_files()
+        self._update_dockerfile()
+
+    def _get_yum_repos(self, image_id):
+        # TODO: This needs a retry
+        dest_path = os.path.join(self.workdir, 'original', image_id)
+        try:
+            os.makedirs(dest_path)
+        except OSError:
+            # Ignore if dirs have already been created. Python 2 doesn't suport exists_ok param
+            pass
+        container_id = self.tasker.copy_file(
+            image_id,
+            src_path='/etc/yum.repos.d',
```

Comment:
Wonder whether a trailing slash ('/etc/yum.repos.d/') would avoid us needing to search within ./yum.repos.d in _parse_yum_repos?